bentoml/_internal/yatai_client/schemas.py

```diff
@@ -0,0 +1,240 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import Optional, List, Dict, Any
+
+from dataclasses_json import DataClassJsonMixin
+from marshmallow import fields
+from dateutil.parser import parse
+
+
+time_format = '%Y-%m-%d %H:%M:%S.%f'
+
+
+def datetime_encoder(time_obj: Optional[datetime]) -> Optional[str]:
+    if not time_obj:
+        return None
+    return time_obj.strftime(time_format)
+
+
+def datetime_decoder(datetime_str: Optional[str]) -> Optional[datetime]:
+    if not datetime_str:
+        return None
+    return parse(datetime_str)
+
+
+datetime_field_kwargs = dict(metadata={'dataclasses_json': {
+    'encoder': datetime_encoder,
+    'decoder': datetime_decoder,
+    'mm_field': fields.DateTime(format='iso')
+}})
+
+
+@dataclass
```

Comment:
can we use `attrs` for this?  the rest of the framework has decided on attrs, and cattrs can replace `dataclasses_json`

Reply:
Well, that's why I said bentoml would be confusing for new developers, because now bentoml has both dataclass and attrs